test/test_general.cpp

```diff
@@ -52,3 +53,65 @@ TEST(General, TestLineParsing) {
   EXPECT_EQ(data[1], "Entry2");
   EXPECT_EQ(data[2], "EscapedEntry1;EscapedEntry2");
 }
+
+TEST(General, Dijkstra) {
+  /*
+            (6)
+        .----------.
+       \/          \/
+  0 <-> 1 --> 2 <-> 3
+    (1)   (2)   (3)
+  */
+  CouplingMap cm = {{0, 1}, {1, 0}, {1, 2}, {2, 3}, {3, 2}, {1, 3}, {3, 1}};
+  Matrix      edgeWeights = {
+      {0, 1, 0, 0}, {1, 0, 2, 6}, {0, 15, 0, 3}, {0, 6, 3, 0}};
+  Matrix distanceTable{};
+  Dijkstra::buildTable(4, cm, distanceTable, edgeWeights,
+                       [](const Dijkstra::Node& n) { return n.cost; });
+  EXPECT_EQ(distanceTable.size(), 4);
+  EXPECT_EQ(distanceTable[0].size(), 4);
+  EXPECT_EQ(distanceTable[0][0], 0);
+  EXPECT_EQ(distanceTable[0][1], 1);
+  EXPECT_EQ(distanceTable[0][2], 3);
+  EXPECT_EQ(distanceTable[0][3], 6);
+  EXPECT_EQ(distanceTable[1].size(), 4);
+  EXPECT_EQ(distanceTable[1][0], 1);
+  EXPECT_EQ(distanceTable[1][1], 0);
+  EXPECT_EQ(distanceTable[1][2], 2);
+  EXPECT_EQ(distanceTable[1][3], 5);
+  EXPECT_EQ(distanceTable[2].size(), 4);
+  EXPECT_EQ(distanceTable[2][0], 10);
+  EXPECT_EQ(distanceTable[2][1], 9);
+  EXPECT_EQ(distanceTable[2][2], 0);
+  EXPECT_EQ(distanceTable[2][3], 3);
+  EXPECT_EQ(distanceTable[3].size(), 4);
+  EXPECT_EQ(distanceTable[3][0], 7);
+  EXPECT_EQ(distanceTable[3][1], 6);
+  EXPECT_EQ(distanceTable[3][2], 3);
+  EXPECT_EQ(distanceTable[3][3], 0);
```

Comment:
I am not quite sure whether googletest is capable of that, but couldn't you write this in a much more compact fashion by specifying a `targetTable` (just like the edge weights matrix) and do `EXPECT_EQ(distanceTable, targetTable);`?

Reply:
I did not think that googletest supports deep collection comparisons, but it actually works. Thank you for the suggestion. The code is much cleaner now that way.